aws/cloudformation/standalone/gen_ai_curriculum/deploy_gen_ai_curriculum_stack

```diff
@@ -0,0 +1,178 @@
+  unless options[:environment]
+    print "Environment (#{VALID_ENVIRONMENTS.join(' or ')}): "
+    options[:environment] = gets.chomp
+  end
+
+  unless VALID_ENVIRONMENTS.include?(options[:environment])
+    puts "Error: Invalid environment '#{options[:environment]}'. Must be #{VALID_ENVIRONMENTS.join(' or ')}."
+    exit 1
+  end
+
+  # Prompt for dry-run if not provided
+  unless options.key?(:dry_run)
+    print "Dry-run (create template only)? (y/n): "
+    response = gets.chomp.downcase
+    options[:dry_run] = response == 'y'
+  end
+
+  # Prompt for image URI if not provided
+  unless options.key?(:image_uri)
+    puts "\nHuggingFace Image URI Details".colorize(:light_yellow)
+    puts "-- Image URI version: " + HUGGINGFACE_IMAGE_URI_VERSION.colorize(:light_cyan)
+    puts "-- Image URI: " + HUGGINGFACE_IMAGE_URI.colorize(:light_cyan)
+    puts "\nTo generate a new image URI, use the #{'get_huggingface_llm_image_uri'.colorize(:light_blue)} Python API from the Sagemaker Python SDK. For example:"
+    puts "\n\t" + "from sagemaker.huggingface import get_huggingface_llm_image_uri".colorize(:light_white)
+    puts "\t" + "image_uri = get_huggingface_llm_image_uri(\"huggingface\", version=\"2.2.0\")".colorize(:light_white)
+    print "\nUse a custom image URI (blank to proceed with current): "
+    response = gets.chomp
+    image_uri = response.empty? ? HUGGINGFACE_IMAGE_URI : response
+  end
+
+  # Fetch token from secrets manager if not specified
+  unless options[:token]
+    print "\nFetching HuggingFace Access Token..."
+    secrets_manager = Aws::SecretsManager::Client.new
+    value = secrets_manager.get_secret_value(
+      {secret_id: "#{options[:environment]}/gen-ai-curriculum/huggingface_token"}
+    )
+    options[:token] = value.secret_string
+    puts "done".colorize(:light_green)
+  end
+
+  puts "\nHuggingFace Image URI: " + image_uri.to_s.colorize(:light_cyan)
+  puts "Environment: " + (options[:environment]).to_s.colorize(:light_cyan)
+  puts "Dry-run mode: " + (options[:dry_run]).to_s.colorize(:light_cyan)
+
+  unless options[:dry_run]
+    puts "\nChecking autoscaling targets...\n"
+    Config::ENDPOINT_CONFIGS.each do |config|
+      deregister_auto_scaling(config[:model_id], options[:environment])
+    end
+  end
+
+  puts "\nTransforming ERB to YAML..."
+  ` erb -T - environment=#{options[:environment]} -r ./config.rb gen_ai_curriculum.yml.erb > gen_ai_curriculum.yml `
+  return unless $?&.exitstatus == 0
+  puts "Success!".colorize(:light_green)
+
+  puts "\nGetting AWS Account ID..."
+  identity = `aws sts get-caller-identity`
+  return unless $?&.exitstatus == 0
+  account_id = JSON.parse(identity)['Account']
+  puts "Using AWS Account ID: #{account_id.colorize(:light_cyan)}"
+
+  cmd = "aws cloudformation deploy " \
+    "--stack-name gen-ai-curriculum-#{options[:environment]} " \
+    "--template-file gen_ai_curriculum.yml " \
+    "--role-arn arn:aws:iam::#{account_id}:role/admin/CloudFormationService " \
```

Comment:
@sureshc helped me figure out that we need to pass this role ARN when deploying the stack to get the right permissions. The `account_id` is fetched above using `aws sts get-caller-identity`